x-pack/plugins/security_solution/public/resolver/view/clickthrough.test.tsx

```diff
@@ -208,6 +211,228 @@ describe('Resolver, when analyzing a tree that has no ancestors and 2 children',
   });
 });
 
+describe('Resolver, when using a generated tree with 20 generations, 4 children per child, and 10 ancestors', () => {
+  beforeEach(async () => {
+    const { metadata: dataAccessLayerMetadata, dataAccessLayer } = generateTreeWithDAL({
```

Comment:
Yeah, we chose to use the generator for a couple reasons:

1. We wanted to generate a fairly large graph to increase our chances of having one of the nodes outside of the bounding box when the graph initially loads. This is required so that we can test the node data states. Generating an arbitrarily large graph was easier to do using the existing generator code in our minds.
2. The data access layer mocks duplicate a lot of functionality for creating fake endpoint events. Whenever we need to change the interface for the data access layer we'll have to update all the mocks function signatures at a minimum. If we can refactor those mocks to use the generator code we created directly or remove them entirely it'll save us having to maintain 6+ places for the data access layer interface.
3. Overall I think having the ability to easily generate a resolver graph instead of creating each node by hand save us from having to write boilerplate code.